### PR TITLE
Describe association between Sampler and TracerProvider.

### LINKED ISSUE #781: Spec needs to mention that a Sampler is associated with a TracerProvider
As discussed in the SIG meeting, this issue is split from #634.

The current spec on sampling ([time of issue creation](https://github.com/open-telemetry/opentelemetry-specification/blob/bc272ab/specification/trace/sdk.md#sampling); [current](https://github.com/open-telemetry/opentelemetry-specification/blob/master/specification/trace/sdk.md#sampling)) specifies the the Sampler interface and [when describing the return value of ShouldSample](https://github.com/open-telemetry/opentelemetry-specification/blob/bc272ab52869df2145065d55300d721ff008e34b/specification/trace/sdk.md#shouldsample) it also (a bit indirectly) describes what the SDK should do in reaction to the sampler. However, the spec needs to describe how a Sampler is associated with a TracerProvider. Also, the definition of what the default sampler is (currently at https://github.com/open-telemetry/opentelemetry-specification/blob/bc272ab52869df2145065d55300d721ff008e34b/specification/trace/sdk.md#built-in-samplers) should move there.

Assigned @alolita as per https://github.com/open-telemetry/opentelemetry-specification/issues/634#issuecomment-672055000

### PR DESCRIPTION
**Pull request by Oberon00**
Fixes #781.

Note that how span and trace IDs are (not) to be generated is currently unspecified, see #864 

## Changes

* Describes association of Sampler with TracerProvider. This is hopefully how everyone already implemented this.

### CODE REVIEW

File: specification/trace/sdk.md [hunk 1]
```diff
@@ -56,7 +58,10 @@ The following table summarizes the expected behavior for each combination of
 | false         | false          | false                         | false                        |
 
 The SDK defines the interface [`Sampler`](#sampler) as well as a set of
-[built-in samplers](#built-in-samplers).
+[built-in samplers](#built-in-samplers) and associates a `Sampler` with each [`TracerProvider`].
+
+When asked to create a Span, the SDK MUST query the `Sampler`'s [`ShouldSample`](#shouldsample) method before actually creating the span, and act accordingly
+(see description of [`ShouldSample`'s](#shouldsample) return value below for how to set `IsRecording` and `Sampled` on the Span, and the [table above](#recording-sampled-reaction-table) on whether to pass the `Span` to `SpanProcessor`s.  A non-recording span MAY be implemented using the same mechanism as when a `Span` is created with no API-implementation installed (`NoOpSpan` or `DefaultSpan`).
```
Comment (carlosalberto) [inline specification/trace/sdk.md:64]: Missing ")".

**Review by carlosalberto — APPROVED** [review]

**Comment by carlosalberto** [comment]
cc @bogdandrutu (as in Java we store this in a `TraceConfig` object that belongs to `TracerProvider` ;) )

**Comment by bogdandrutu** [comment]
Yes, but in Java the config is per TP so matches this.

**Review by bogdandrutu — APPROVED** [review]

**Review by arminru — APPROVED** [review]

**Comment by carlosalberto** [comment]
This looks ready to merge. Please update @Oberon00

**Pull request merged**